wger/weight/helpers.py

```diff
@@ -43,21 +43,36 @@ def parse_weight_csv(request, cleaned_data):
     # csv.reader expects a file-like object, so use StringIO
     parsed_csv = csv.reader(six.StringIO(cleaned_data['csv_input']),
                             dialect)
-
+    distinct_weight_entries = []
     weight_list = []
     error_list = []
+
+    # Process the CSV items first
     for row in parsed_csv:
         try:
             parsed_date = datetime.datetime.strptime(row[0], cleaned_data['date_format'])
             parsed_weight = decimal.Decimal(row[1].replace(',', '.'))
-
-            weight_list.append(WeightEntry(date=parsed_date,
-                                           weight=parsed_weight,
-                                           user=request.user))
+            duplicate_date_in_db = WeightEntry.objects.filter(date=parsed_date,
+                                                              user=request.user)
```

Comment:
Roland, I just had a chance to look at it now. Your point makes sense. I
made the change accordingly.


On Wed, Jul 1, 2015 at 10:14 AM, Roland Geider <notifications@github.com>
wrote:

> In wger/weight/helpers.py
> <https://github.com/rolandgeider/wger/pull/153#discussion_r33681907>:
>
> >      for row in parsed_csv:
> >          try:
> >              parsed_date = datetime.datetime.strptime(row[0], cleaned_data['date_format'])
> >              parsed_weight = decimal.Decimal(row[1].replace(',', '.'))
> > -
> > -            weight_list.append(WeightEntry(date=parsed_date,
> > -                                           weight=parsed_weight,
> > -                                           user=request.user))
> > +            duplicate_date_in_db = WeightEntry.objects.filter(date=parsed_date,
> > +                                                              user=request.user)
>
> @max111 <https://github.com/max111> Did you get a chance to look at this?
> :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rolandgeider/wger/pull/153/files#r33681907>.
>